### PR TITLE
fix panic in provider server while onboarding

### PR DESCRIPTION
**Pull request by rewantsoni**
Fix provider server panic while onboarding Consumers when the value of the storageQuota in the onboardingTicket is set to nil

Introduced with: https://github.com/red-hat-storage/ocs-operator/pull/2669

### CODE REVIEW

**Review by leelavg — COMMENTED** [review]
pls mention which PR introduced this, not able to recall, thanks.

File: services/provider/server/consumer.go [hunk 1]
```diff
@@ -74,8 +74,13 @@ func (c *ocsConsumerManager) Create(ctx context.Context, onboard ifaces.StorageC
 			},
 		},
 		Spec: ocsv1alpha1.StorageConsumerSpec{
-			Enable:            false,
-			StorageQuotaInGiB: storageQuotaInGiB,
+			Enable: false,
+			StorageQuotaInGiB: func() int {
+				if storageQuotaInGiB != nil {
+					return int(*storageQuotaInGiB)
```
Comment (leelavg) [inline services/provider/server/consumer.go:80]: converting uint to int here, yes, we may not have larger values where we overflow but why not using `uint(*storageQuotaInGiB)`?

Comment (leelavg) [reply services/provider/server/consumer.go:80]: ouch, I think there was some discussion which made this field in token struct as uint, seems non consistent with only a benefit of reducing one extra check for signedness.

Anyways, can't change that now.

Comment (nb-ohad) [reply services/provider/server/consumer.go:80]: What is the issue with converting the unit to int? We are not expecting very large numbers here.
How does moving the conversion from outside the function into the function solves the problem?

**Review by nb-ohad — CHANGES_REQUESTED** [review]

**Comment by rewantsoni** [comment]
>What is the issue with converting the unit to int? We are not expecting very large numbers here.
How does moving the conversion from outside the function into the function solves the problem?

@nb-ohad The issue is with converting the pointer to unit to int. If the storageQuotaInGib is nil it will panic. I moved everything inside a inline function to make it clean and return 0 as storageQuotaInGib if the value is nil (we could have the same check outside the function as well)

**Comment by nb-ohad** [comment]
@rewantsoni You should not convert a pointer. You should convert the value is not the same thing. 
Converting a pointer means that you are trying to reference the same underlying value but at the same time interpreting it as something else. That is wrong in this case. 

You should convert the value (outside of the function), and send the converted value into the function.

**Comment by nb-ohad** [comment]
A good fix for the bug will be
```
if size := onboardingTicket.StorageQuotaInGiBl size == nil || *size > math.MaxInt {
    // return some error
} 
storageConsumerUUID, err := s.consumerManager.Create(ctx, req, int(*size))

**Pull request merged**